packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -215,9 +215,16 @@ class OverlayState extends State<Overlay> {
   }
 
   void _remove(OverlayEntry entry) {
-    setState(() {
-      _entries.remove(entry);
+    assert(() {
+      if (!mounted)
+        assert(_entries.remove(entry));
```

Comment:
Ok.  I guess this is fine.  Maybe add a comment to the declaration of _entries warning future developers that it is mutated in checked mode differently than in release mode.